src/components/BookForm.js

```diff
@@ -0,0 +1,79 @@
+import React, { useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { addBook } from '../redux/books/booksSlice';
+
+function BookForm() {
+  const [title, setTitle] = useState('');
+  const [author, setAutor] = useState('');
+  const [category, setCategory] = useState('Fiction');
+  const bookstore = useSelector((state) => state.books.bookstore);
+  const categories = useSelector((state) => state.category.categories);
+  const dispath = useDispatch();
+  const addBookHandler = (e) => {
+    if (title !== '' && author !== '') {
+      e.preventDefault();
+      const book = {
+        item_id: `item${bookstore.length + 1}`,
```

Comment:
- [ ] I don't think it's a good idea, in this case, to use the array length for IDs considering that books be removed, so at some point, multiple books may have the same ID. The ID won't be unique. I would recommend you use a more unique source of IDs like the `uuid` package.

  For example, you have three initial books. Remove one and another book. Delete the book you added you will see that the two books are removed.